manifests/server.pp

```diff
@@ -89,79 +91,80 @@
-  $postgresql_conf_path                            = $postgresql::params::postgresql_conf_path,
-  Optional[Stdlib::Filemode] $postgresql_conf_mode = $postgresql::params::postgresql_conf_mode,
-  $recovery_conf_path                              = $postgresql::params::recovery_conf_path,
-
-  $datadir                                         = $postgresql::params::datadir,
-  $xlogdir                                         = $postgresql::params::xlogdir,
-  $logdir                                          = $postgresql::params::logdir,
-
-  $log_line_prefix                                 = $postgresql::params::log_line_prefix,
-
-  $pg_hba_conf_defaults                            = $postgresql::params::pg_hba_conf_defaults,
-
-  $user                                            = $postgresql::params::user,
-  $group                                           = $postgresql::params::group,
-
-  $needs_initdb                                    = $postgresql::params::needs_initdb,
-
-  $encoding                                        = $postgresql::params::encoding,
-  $locale                                          = $postgresql::params::locale,
-  $data_checksums                                  = $postgresql::params::data_checksums,
-  $timezone                                        = $postgresql::params::timezone,
-
-  $manage_pg_hba_conf                              = $postgresql::params::manage_pg_hba_conf,
-  $manage_pg_ident_conf                            = $postgresql::params::manage_pg_ident_conf,
-  $manage_recovery_conf                            = $postgresql::params::manage_recovery_conf,
-  Boolean $manage_postgresql_conf_perms            = $postgresql::params::manage_postgresql_conf_perms,
-  Boolean $manage_selinux                          = $postgresql::params::manage_selinux,
-  $module_workdir                                  = $postgresql::params::module_workdir,
-
-  $manage_datadir                                  = $postgresql::params::manage_datadir,
-  $manage_logdir                                   = $postgresql::params::manage_logdir,
-  $manage_xlogdir                                  = $postgresql::params::manage_xlogdir,
-  $password_encryption                             = $postgresql::params::password_encryption,
-  $extra_systemd_config                            = $postgresql::params::extra_systemd_config,
-
-  Hash[String, Hash] $roles              = {},
-  Hash[String, Any] $config_entries      = {},
-  Postgresql::Pg_hba_rules $pg_hba_rules = {},
-
-  Boolean $backup_enable = $postgresql::params::backup_enable,
-  Hash $backup_options = {},
-  Enum['pg_dump'] $backup_provider = $postgresql::params::backup_provider,
+  String[1]                                    $package_name                 = $postgresql::params::server_package_name,
+  Enum['present', 'absent', 'latest']          $package_ensure               = $postgresql::params::package_ensure,
```

Comment:
This will not work with versions, also affected: all other package resources

Reply:
Yeah, brought this topic up in a short discussion and figured out that my definition of $ensure was too limiting. Specially in comparison to official documentation. Will be updating all affected $ensure values to reflect better the valid values.

Reply:
Note: regarding the context PR in stdlib... while we agree that the full Enum is, essentially, an alias for String[1], we still want to keep $ensure datatypes as Enums. The main reason here is to keep documentation as accurate as possible in the long run.

It also makes the variable easier to identify and track when looking at the code. And, of course, it can also be helpful when debugging.

Reply:
Done